File: Lecture16: Selection Sort/selectionSort.java
import java.util.*;

public static void selectionSort(int[] arr, int n)
{   
    for(int i = 0; i < n-1; i++ ) {
        int minIndex = i;
        
        for(int j = i+1; j<n; j++) {
            
            if(arr[j] < arr[minIndex]) 
                minIndex = j;
            
        }
        swap(arr,minIndex,i);
    }
}

public void swap(int[] arr,int i,int j){
  int temp;
  
  temp = arr[i];
  arr[i] = arr[j];
  arr[j] = temp;
  
}

File: Lecture016 Selection Sort/SelectionSort.py
def selectionSort(arr, n):
    for i in range(n-1):
        minIndex = i
        
        for j in range (i+1,n):
            if arr[j] < arr[minIndex]:
                minIndex = j
            
        arr[minIndex] , arr[i]=arr[i] , arr[minIndex]
        
    return arr
  
if __name__ == "__main__":
    n=5 # size of arr
    arr=[5,9,2,4,3] #take input in list 
    
    ans = selectionSort (arr,n)
    print(ans)
